src/ng/parse.js

```diff
@@ -426,375 +574,1027 @@ Parser.prototype = {
+    if (computed) return this.computedMember(left, right);
+    return this.nonComputedMember(left, right);
   },
 
-  equality: function() {
-    var left = this.relational();
-    var token;
-    while ((token = this.expect('==','!=','===','!=='))) {
-      left = this.binaryFn(left, token.text, this.relational());
-    }
-    return left;
+  addEnsureSafeObject: function(item) {
+    this.current().body.push(this.ensureSafeObject(item), ';');
   },
 
-  relational: function() {
-    var left = this.additive();
-    var token;
-    while ((token = this.expect('<', '>', '<=', '>='))) {
-      left = this.binaryFn(left, token.text, this.additive());
-    }
-    return left;
+  addEnsureSafeMemberName: function(item) {
+    this.current().body.push(this.ensureSafeMemberName(item), ';');
   },
 
-  additive: function() {
-    var left = this.multiplicative();
-    var token;
-    while ((token = this.expect('+','-'))) {
-      left = this.binaryFn(left, token.text, this.multiplicative());
-    }
-    return left;
+  addEnsureSafeFunction: function(item) {
+    this.current().body.push(this.ensureSafeFunction(item), ';');
   },
 
-  multiplicative: function() {
-    var left = this.unary();
-    var token;
-    while ((token = this.expect('*','/','%'))) {
-      left = this.binaryFn(left, token.text, this.unary());
-    }
-    return left;
+  ensureSafeObject: function(item) {
+    return 'ensureSafeObject(' + item + ',text)';
   },
 
-  unary: function() {
-    var token;
-    if (this.expect('+')) {
-      return this.primary();
-    } else if ((token = this.expect('-'))) {
-      return this.binaryFn(Parser.ZERO, token.text, this.unary());
-    } else if ((token = this.expect('!'))) {
-      return this.unaryFn(token.text, this.unary());
-    } else {
-      return this.primary();
-    }
+  ensureSafeMemberName: function(item) {
+    return 'ensureSafeMemberName(' + item + ',text)';
   },
 
-  fieldAccess: function(object) {
-    var getter = this.identifier();
+  ensureSafeFunction: function(item) {
+    return 'ensureSafeFunction(' + item + ',text)';
+  },
 
-    return extend(function $parseFieldAccess(scope, locals, self) {
-      var o = self || object(scope, locals);
-      return (o == null) ? undefined : getter(o);
-    }, {
-      assign: function(scope, value, locals) {
-        var o = object(scope, locals);
-        if (!o) object.assign(scope, o = {});
-        return getter.assign(o, value);
-      }
-    });
+  lazyRecurse: function(ast, intoId, nameId, recursionFn, create) {
+    var self = this;
+    return function() {
+      self.recurse(ast, intoId, nameId, recursionFn, create);
+    };
   },
 
-  objectIndex: function(obj) {
-    var expression = this.text;
+  lazyAssign: function(id, value) {
+    var self = this;
+    return function() {
+      self.assign(id, value);
+    };
+  },
 
-    var indexFn = this.expression();
-    this.consume(']');
+  stringEscapeRegex: new RegExp('[^ a-zA-Z0-9]', 'g'),
```

Comment:
Normally regexps are declared using the `/[^ a-zA-Z0-9]/g` style. Any reason not to here?

This will also escape a lot more then necessary, but a bigger/uglier regex might not be worth it? But `/[ -&(-}]/g` (smaller/uglier!) would match all standard ascii except the quote...